azkaban-common/src/main/java/azkaban/executor/ExecutionFlowDao.java

```diff
@@ -255,17 +255,31 @@ private void updateExecutableFlow(final ExecutableFlow flow, final EncodingType
             + "SET status=?,update_time=?,start_time=?,end_time=?,enc_type=?,flow_data=? "
             + "WHERE exec_id=?";
 
-    final String json = JSONUtils.toJSON(flow.toObject());
     byte[] data = null;
     try {
+      // If this action fails, the execution must be failed.
+      final String json = JSONUtils.toJSON(flow.toObject());
```

Comment:
I believe flow.toObject() method is throwing NPE if SLA option list (i.e. this.executionOptions.getSlaOptions()) is null or one of the value in the list (i.e. this.executionOptions.getSlaOptions()) is null. If that is the case we could fix root cause of NPE in the ExecutableFlow.toObject() method. The corresponding code to populate SLA options can be modified as follows.
   final List<Map<String, Object>> slaOptions =
        Optional.ofNullable(this.executionOptions.getSlaOptions()).orElse(Collections.emptyList()).stream()
        .filter(Objects::nonNull)
        .map(slaOption -> slaOption.toObject())
        .collect(Collectors.toList());

Reply:
Thanks! It was supposed to be done and I completely forgot!
I did not do it earlier to test if the catch logic works properly and was supposed to work on it after that.
Good catch!